pkg/server/router/credential.go

```diff
@@ -37,6 +37,65 @@ func NewCredentialRouter(s svcframework.Service) (*CredentialRouter, error) {
 	return &CredentialRouter{service: credService}, nil
 }
 
+type BatchCreateCredentialsRequest struct {
+	// Required. The list of create credential requests. Cannot be more than 1000 items.
+	Requests []CreateCredentialRequest `json:"requests" maxItems:"1000" validate:"required,dive"`
+}
+
+func (r BatchCreateCredentialsRequest) toServiceRequest() credential.BatchCreateCredentialsRequest {
+	var req credential.BatchCreateCredentialsRequest
+	for _, routerReq := range r.Requests {
+		req.Requests = append(req.Requests, routerReq.toServiceRequest())
+	}
+	return req
+}
+
+type BatchCreateCredentialsResponse struct {
+	// The credentials created.
+	Credentials []credmodel.Container `json:"credentials"`
+}
+
+// BatchCreateCredentials godoc
+//
+//	@Summary		Batch Create Credentials
+//	@Description	Create a batch of verifiable credentials.
+//	@Tags			CredentialAPI
+//	@Accept			json
+//	@Produce		json
+//	@Param			request	body		BatchCreateCredentialsRequest	true	"The batch requests"
+//	@Success		201		{object}	BatchCreateCredentialsResponse
+//	@Failure		400		{string}	string	"Bad request"
+//	@Failure		500		{string}	string	"Internal server error"
+//	@Router			/v1/credentials/batchCreate [put]
```

Comment:
gut level this feels off and not very resutful

what about `v1/credentials/batch` ?

Reply:
still should be clear since it's namespaced to credentials and follows PUT for rest semantics

Reply:
yes, on quick google:
- https://www.mscharhag.com/api-design/bulk-and-batch-operations
- https://www.codementor.io/blog/batch-endpoints-6olbjay1hd
note this one shows a google API `POST https://www.googleapis.com/batch/drive/v3`
- https://www.ibm.com/docs/en/was-liberty/base?topic=liberty-batch-rest-api
- https://www.servicenow.com/community/now-platform-articles/an-introduction-to-the-batch-rest-api-endpoint-api-now-batch/ta-p/2317136

Reply:
the examples all have different approaches. None use a `<resource>/batch` approach. Are you suggesting to have a top level `v1/batch` endpoint? 
* https://www.mscharhag.com/api-design/bulk-and-batch-operations suggest using a `v1/batch` endpoint, without the resources it's acting on. It supports batching of many things. 
* https://www.codementor.io/blog/batch-endpoints-6olbjay1hd also showcases an example to an endpoint that has the verb on it `/api/v2/users/create_or_update_many.json`. The drive endpoint uses the approach of having a 
* https://www.ibm.com/docs/en/was-liberty/base?topic=liberty-batch-rest-api uses `batch` to namespace all batch requests `/ibm/api/batch/jobinstances`
* https://www.servicenow.com/community/now-platform-articles/an-introduction-to-the-batch-rest-api-endpoint-api-now-batch/ta-p/2317136 again usese the `batch` namespace approach.

Reply:
I'm ok with the batch namespace approach, my main gripe is that `batch` is more of a resource than `batchCreate` which is an RPC, and we should be as RESTful as possible

Reply:
another option is no batch endpoint, but a batch flag or request parameter